components/src/core/Drawer/DrawerHeader.tsx

```diff
@@ -56,90 +67,94 @@ export type DrawerHeaderProps = ToolbarProps & {
+>(({ backgroundColor, fontColor, theme }) => ({
+    paddingRight: 0,
+    paddingLeft: 0,
+    width: '100%',
+    alignItems: 'center',
+    boxSizing: 'border-box',
+    minHeight: `4rem`,
+    [theme.breakpoints.down('sm')]: {
+        minHeight: `3.5rem`,
+    },
+    backgroundColor:
+        backgroundColor || (theme.palette.mode === 'dark' ? theme.palette.primary.dark : theme.palette.primary.main),
+    color:
+        fontColor ||
+        theme.palette.getContrastText(
+            backgroundColor || (theme.palette.mode === 'dark' ? theme.palette.primary.dark : theme.palette.primary.main)
+        ),
+    [`& .${drawerHeaderClasses.nonCLickable}`]: {},
+    [`& .${drawerHeaderClasses.railIcon}`]: {
+        marginLeft: theme.spacing(0.5),
+        minWidth: 'calc(3.5rem + 16px)',
+        justifyContent: 'center',
+        '&$nonClickable': {
+            marginLeft: 0,
         },
-        subtitle: {
-            lineHeight: '1.2rem', // Anything lower than 1.2rem cuts off bottom text of 'g' or 'y'.
-            marginTop: '-0.125rem',
-        },
-        title: {
-            fontWeight: 600,
-            lineHeight: '1.6rem', // Anything lower than 1.6rem cuts off bottom text of 'g' or 'y'.
-        },
-    })
-);
+    },
+}));
+
+const Background = styled(Box, { name: 'drawer-header', slot: 'background' })<
+    Pick<DrawerHeaderProps, 'backgroundImage' | 'backgroundOpacity'>
+>(({ backgroundImage, backgroundOpacity }) => ({
+    position: 'absolute',
+    zIndex: 0,
+    width: '100%',
+    backgroundSize: 'cover',
+    height: '100%',
+    backgroundPosition: 'center',
+    backgroundImage: `url(${backgroundImage})`,
+    opacity: backgroundOpacity,
+}));
+
+const Navigation = styled(Box, { name: 'drawer-header', slot: 'navigation' })(({ theme }) => ({
+    marginLeft: theme.spacing(2),
+    minWidth: '2.5rem',
+    height: '100%',
+    display: 'flex',
+    alignItems: 'center',
+    zIndex: 1,
+}));
+
+const Content = styled(Box, { name: 'drawer-header', slot: 'content' })(({ theme }) => ({
+    marginLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    minHeight: '4rem',
+    display: 'flex',
+    justifyContent: 'center',
+    alignSelf: 'stretch',
+    flexDirection: 'column',
+    width: 'calc(100% - 2.5rem - 32px)',
+    boxSizing: 'border-box',
+    zIndex: 1,
+    [theme.breakpoints.down('sm')]: {
+        minHeight: `3.5rem`,
+    },
+}));
+
+const Title = styled(Typography, { name: 'drawer-header', slot: 'title' })(() => ({
+    fontWeight: 600,
+    lineHeight: '1.6rem', // Anything lower than 1.6rem cuts off bottom text of 'g' or 'y'.
+}));
+
+const Subtitle = styled(Typography, { name: 'drawer-header', slot: 'subtitle' })(() => ({
+    lineHeight: '1.2rem', // Anything lower than 1.2rem cuts off bottom text of 'g' or 'y'.
+    marginTop: '-0.125rem',
+}));
+
+const NonClickableIcon = styled(Box, { name: 'drawer-header', slot: 'nonClickableIcon' })(() => ({
```

Comment:
I prefer this to be named using kebab case for these classnames 

The generated classnames make sense as camelCase since they will be used with object syntax, while these styled component global styles will be set using css selectors
```suggestion
const NonClickableIcon = styled(Box, { name: 'drawer-header', slot: 'non-clickable-icon' })(() => ({
```